Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp

```diff
@@ -262,6 +262,34 @@ Duration* create_negated_temporal_duration(GlobalObject& global_object, Duration
     return MUST(create_temporal_duration(global_object, -duration.years(), -duration.months(), -duration.weeks(), -duration.days(), -duration.hours(), -duration.minutes(), -duration.seconds(), -duration.milliseconds(), -duration.microseconds(), -duration.nanoseconds()));
 }
 
+// 7.5.9 CalculateOffsetShift ( relativeTo, y, mon, w, d, h, min, s, ms, mus, ns ), https://tc39.es/proposal-temporal/#sec-temporal-calculateoffsetshift
+ThrowCompletionOr<double> calculate_offset_shift(GlobalObject& global_object, Value relative_to, double years, double months, double weeks, double days, double hours, double minutes, double seconds, double milliseconds, double microseconds, double nanoseconds)
+{
+    // 1. If Type(relativeTo) is not Object or relativeTo does not have an [[InitializedTemporalZonedDateTime]] internal slot, return 0.
+    if (!relative_to.is_object() || !is<ZonedDateTime>(relative_to.as_object()))
+        return 0.0;
+
+    auto& relative_to_zoned_date_time = static_cast<ZonedDateTime&>(relative_to.as_object());
```

Comment:
In the case where the thing is used throughout most of the AO after having confirmed it's of a certain type, I usually invert this and suffix the initial variable with `_value`, so that I can use the spec's for the most part.

i.e.

```suggestion
    auto& relative_to = static_cast<ZonedDateTime&>(relative_to_value.as_object());
```